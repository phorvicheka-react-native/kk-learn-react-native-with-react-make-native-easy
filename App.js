/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, {Component} from 'react';
import Router from './app/routes';

class App extends Component {
  render () {
    return (
      <Router />
    );
  }
}

export default App;
