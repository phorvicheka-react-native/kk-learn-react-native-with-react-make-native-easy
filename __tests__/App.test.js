import 'react-native';
/* import React from 'react';
import App from '../App.js'; */

// Note: test renderer must be required after react-native.
/* import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import offlineStorage from '../app/utils/offline.utils.js';
import mockAsyncStorage from '@react-native-community/async-storage/jest/async-storage-mock';

const mockStore = configureStore([]);

jest.mock('@react-native-community/async-storage', () => mockAsyncStorage);
import AsyncStorage from '@react-native-community/async-storage'; */

describe('My Connected React-Redux Component', () => {
  /* let store;
  let component;
 
  beforeEach(() => {
    store = mockStore({
      test: 'sample text',
      content: {
        title: 'sample title',
        text: 'sample text'
      }
    });
 
    component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    ).toJSON();
  }); */
 
  it('should render with given state from Redux store', () => {    
    /* expect(component).toMatchSnapshot();
    AsyncStorage.getItem = jest.fn();
    AsyncStorage.getItem(offlineStorage.keys.NOTES);
    expect(AsyncStorage.getItem).toBeCalledWith(offlineStorage.keys.NOTES); */
  });
});