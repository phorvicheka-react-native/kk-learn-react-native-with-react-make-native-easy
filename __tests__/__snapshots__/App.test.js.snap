// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          false,
        ]
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderBottomColor": "#a7a7aa",
                "flex": 1,
                "shadowColor": "#a7a7aa",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.9)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                Home
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "column",
                          "justifyContent": "space-between",
                          "paddingTop": 20,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "alignSelf": "flex-start",
                            "fontSize": 14,
                            "fontWeight": "700",
                            "padding": 10,
                          }
                        }
                      >
                         Note Title
                      </Text>
                      <TextInput
                        allowFontScaling={true}
                        onChangeText={[Function]}
                        rejectResponderTermination={true}
                        style={
                          Object {
                            "alignSelf": "stretch",
                            "backgroundColor": "#ebe9e9",
                            "padding": 10,
                          }
                        }
                        underlineColorAndroid="transparent"
                        value="sample title"
                      />
                      <Text
                        style={
                          Object {
                            "alignSelf": "flex-start",
                            "fontSize": 14,
                            "fontStyle": "italic",
                            "fontWeight": "200",
                            "padding": 10,
                          }
                        }
                      >
                         Please type your note below 
                      </Text>
                      <TextInput
                        allowFontScaling={true}
                        multiline={true}
                        onChangeText={[Function]}
                        rejectResponderTermination={true}
                        style={
                          Object {
                            "alignSelf": "stretch",
                            "backgroundColor": "#ebe9e9",
                            "flex": 1,
                            "fontSize": 14,
                            "fontWeight": "200",
                            "padding": 10,
                            "textAlignVertical": "top",
                          }
                        }
                        underlineColorAndroid="transparent"
                        value="sample text"
                      />
                      <View
                        onLayout={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <Text
                            onPress={[Function]}
                            style={
                              Object {
                                "fontWeight": "700",
                                "padding": 10,
                              }
                            }
                          >
                            Save
                          </Text>
                          <Text
                            style={
                              Object {
                                "fontSize": 12,
                                "padding": 10,
                              }
                            }
                          >
                            11
                             characters
                          </Text>
                        </View>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "white",
                            "opacity": 1,
                            "padding": 15,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontWeight": "700",
                            }
                          }
                        >
                          About Us
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
